common-npm-packages/azure-arm-rest-v2/azure-arm-management-group.ts

```diff
@@ -44,13 +44,14 @@ export class ManagementGroupDeployments extends depolymentsBase.DeploymentsBase
         this.client = client;
     }
 
-    public createOrUpdate(deploymentName, deploymentParameters, callback) {
+    public createOrUpdate(deploymentName, deploymentParameters, callback, isWhatIf: boolean = false) {
 
         // Create HTTP request uri
         var requestUri = this.client.getRequestUri(
-            '//providers/Microsoft.Management/managementGroups/{managementGroupId}/providers/Microsoft.Resources/deployments/{deploymentName}',
+            '//providers/Microsoft.Management/managementGroups/{managementGroupId}/providers/Microsoft.Resources/deployments/{deploymentName}{whatIf}',
```

Comment:
Won't we need to handle the API response appropriately:
1. WhatIf might not return any `outputs` property and hence that should be handled in task in case user has specified a output variable
2. What should we do with whatif's response. Does it need to be printed in task logs so that user can make deployment decision based on that?